test/e2e/opaque_resource.go

```diff
@@ -215,9 +217,65 @@ var _ = framework.KubeDescribe("Opaque resources [Feature:OpaqueResources]", fun
 			return err
 		}
 		predicate = func(e *v1.Event) bool {
-			return e.Type == "Warning" &&
-				e.Reason == "FailedScheduling" &&
-				strings.Contains(e.Message, "failed to fit in any node")
+			return strings.HasPrefix(e.Name, pod.Name) &&
+				e.Reason == "OutOfpod.alpha.kubernetes.io/opaque-int-resource-foo"
```

Comment:
Not after the fix (test needs update). This is what happens when a pod gets bound to a node with less than the necessary resource available. This is what the person who reported the bug saw. The Kubelet sets the pod phase to Failed and this is what shows up in kubectl.

Actually that's a separate question, and a separate issue -- what should happen in that case? Do users expect the pod to bounce off of that node and get scheduled or do they expect this terminal state?